Readme.md

```diff
@@ -0,0 +1,494 @@
+
+
+---------------------
+
+## Read
+
+To read the information from the database, we can use the `find` operator and let's begin by reading a random document.
+
+```kotlin
+val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+collection.find<Restaurant>().limit(1).collect {
+    println(it)
+}
+```
+
+`find` operator returns a list of results but since we are only interested in a single document, we can use `limit` operator in conjunction to
+limit our result set, in this case, it would be a single document.
+
+If we extend this further and want to read a specific document, we can add filter parameters over the top of it like
+
+```kotlin
+val queryParams = Filters
+    .and(
+        listOf(
+            eq("cuisine", "American"),
+            eq("borough", "Queens")
+        )
+    )
+```
+
+Or any of the operators from this [list](https://www.mongodb.com/docs/manual/reference/operator/query/). And final code would look like this.
+
+```kotlin
+suspend fun readSpecificDocument(database: MongoDatabase) {
+    val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+    val queryParams = Filters
+        .and(
+            listOf(
+                eq("cuisine", "American"),
+                eq("borough", "Queens")
+            )
+        )
+
+
+    collection
+        .find<Restaurant>(queryParams)
+        .limit(2)
+        .collect {
+            println(it)
+        }
+
+}
+```
+
+And our output would be something like this
+
+![read specific doc output](https://images.contentstack.io/v3/assets/blt39790b633ee0d5a7/bltd837ac1a039ae43f/648ae83f0f2d9b551eed55e2/Screenshot_2023-06-15_at_11.30.20.png)
+
+Another practical use-case that comes with a read operation is how to add pagination to the results, this can be done with `limit` and
+`offset` operator.
+
+```kotlin
+suspend fun readWithPaging(database: MongoDatabase, offset: Int, pageSize: Int) {
+    val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+    val queryParams = Filters
+        .and(
+            listOf(
+                eq(Restaurant::cuisine.name, "American"),
+                eq(Restaurant::borough.name, "Queens")
+            )
+        )
+
+    collection
+        .find<Restaurant>(queryParams)
+        .limit(pageSize)
+        .skip(offset)
+        .collect {
+            println(it)
+        }
+}
+```
+
+But with this approach, often query response time increase with value of `offset` and to overcome this we can benefit by creating a `Index` as
+shown below.
+
+```kotlin
+val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+val options = IndexOptions().apply {
+    this.name("restaurant_id_index")
+    this.background(true)
+}
+
+collection.createIndex(
+    keys = Indexes.ascending("restaurant_id"),
+    options = options
+)
+```
```

Comment:
Were indexes part of the request specification for this content? I'd leave them out if they were not part of the request, but it is up to you.

Reply:
Indirectly yes.